Fixes #10540 — Vertical bar character | not escaped in cURL (CMD) request snippet body

Divyanshu-Mishra9620 opened the pull request:
## Problem
When generating cURL (CMD) request snippets, the vertical bar character | in the request body was not escaped. In Windows CMD, this causes the string after | to be interpreted as a command, resulting in errors.

## Solution
Updated the CMD snippet generator to escape | as ^| in the request body.
Enabled request snippets in the dev environment for easier testing.
## Testing
Verified that the generated cURL (CMD) snippet now escapes | as ^|.
Confirmed that pasting the snippet in Windows CMD no longer causes command parsing errors. 
## Before
<img width="1587" height="1002" alt="swagger" src="https://github.com/user-attachments/assets/64acd705-91c0-4065-be04-2992d64199eb" />
<img width="698" height="92" alt="swagger2" src="https://github.com/user-attachments/assets/6bd54f7e-5935-4e71-a5c3-259d7676c2ba" />

## After the change=> .replace(/\|/g, "^|")
<img width="503" height="259" alt="Screenshot 2025-10-08 182518" src="https://github.com/user-attachments/assets/d56103e3-64f6-4ac2-86f4-e34c6482c736" />
<img width="905" height="184" alt="Screenshot 2025-10-08 182436" src="https://github.com/user-attachments/assets/c876a256-2a7e-4836-bbcd-f6c0c070a0cc" />

Divyanshu-Mishra9620 committed:
Fix #10540: Escape vertical bar | in cURL (CMD) request snippet body for Windows CMD compatibility

Divyanshu-Mishra9620 committed:
Merge branch 'master' into fix-#10540-Vertical-bar-character-pipe-not-escaped-curl-cmd-snippet

Divyanshu-Mishra9620 committed:
Fix: Escape vertical bar (|) in cURL (CMD) request snippet body and add unit test. Remove unrelated changes.

Divyanshu-Mishra9620 committed:
Merge branch 'fix-#10540-Vertical-bar-character-pipe-not-escaped-curl-cmd-snippet' of https://github.com/Divyanshu-Mishra9620/swagger-ui into fix-#10540-Vertical-bar-character-pipe-not-escaped-curl-cmd-snippet

Divyanshu-Mishra9620 committed:
Merge branch 'master' into fix-#10540-Vertical-bar-character-pipe-not-escaped-curl-cmd-snippet

glowcloud reviewed (commented):
Thanks for the tests! I left a few comments requesting some changes. 

As for the changes to files in the `dist` folder - they are updated only during the release process, as can be seen in the commit history [here](https://github.com/swagger-api/swagger-ui/commits/master/dist).

Divyanshu-Mishra9620 committed:
Update dev-helpers/dev-helper-initializer.js

Co-authored-by: Oliwia Rogala <oliwia.rogala97@gmail.com>

Divyanshu-Mishra9620 committed:
Export escapeCMD function from fn.js

Divyanshu-Mishra9620 committed:
Remove test for vertical bar escaping in bash

Removed test case for escaping vertical bar in bash.

Divyanshu-Mishra9620 committed:
Enable request snippets in Swagger UI configuration

glowcloud reviewed (commented):
Hi @Divyanshu-Mishra9620, 

I requested some last changes:
1. I tested the PR locally and the linting did not pass for the test file. Our style guide prefers using double quotes when possible and avoids semicolons - I left comments related to that.
2. As mentioned before, we do not accept changes to any files in `dist` folder through PRs. Please remove any changes done there.

Divyanshu-Mishra9620 committed:
Fix vertical bar escaping in CMD and apply code style compliance

- Add vertical bar (|) escaping in escapeCMD function
- Update test to use double quotes and no semicolons per project style guide
- Revert dev-helper-initializer.js to previous commented state

Fixes #10540

Divyanshu-Mishra9620 committed:
Resolve merge conflicts and maintain style compliance

Divyanshu-Mishra9620 committed:
Remove dist files and unrelated changes per maintainer feedback

- Remove all changes to dist/ folder as they are not accepted in PRs
- Remove unrelated changes to source files not part of the bug fix
- Keep only the vertical bar escaping fix and its test

Divyanshu-Mishra9620 committed:
Remove dist LICENSE.txt files that shouldn't be in repository

lukaszzazulak commented on src/core/components/responses.jsx:
```diff
@@ -87,11 +87,10 @@ export default class Responses extends React.Component {
     const regionId = createHtmlReadyId(`${method}${path}_responses`)
     const controlId = `${regionId}_select`
 
-    return (!nonExtensionResponses || !nonExtensionResponses.size) ? null : (
```
what was the reason of removing this code ? is this related to the vertical bar character problem ?

lukaszzazulak commented:
> @glowcloud Hey i have fixed, if the issue still exist i will work on that.

@Divyanshu-Mishra9620 It look like there are still changes unrelated to the bug, have a look at my comment for example.

Divyanshu-Mishra9620 committed:
fix: escape vertical bar (|) in CMD request snippets (#10540) and revert unrelated changes

Divyanshu-Mishra9620 commented:
## This PR fixes issue #10540 by ensuring the vertical bar (|) character is properly escaped in CMD request snippets.

* In an earlier commit, I mistakenly changed unrelated logic in the [Responses](vscode-file://vscode-app/c:/Users/divyanshu/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) component (removing the [nonExtensionResponses](vscode-file://vscode-app/c:/Users/divyanshu/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) filter). This was not related to the vertical bar bug and was an oversight. I have now reverted those changes, and this PR only contains the intended bug fix and its related test.

Divyanshu-Mishra9620 committed:
Merge branch 'master' into fix-#10540-Vertical-bar-character-pipe-not-escaped-curl-cmd-snippet
